controllers/activemqartemis_reconciler.go

```diff
@@ -1104,95 +1220,52 @@ func formatTemplatedString(customResource *brokerv1beta1.ActiveMQArtemis, templa
 	}
-	namespacedName := types.NamespacedName{
-		Name:      customResource.Name,
-		Namespace: customResource.Namespace,
+
+	sslSecret := &corev1.Secret{}
+	if err := resources.Retrieve(secretNamespacedName, client, sslSecret); err != nil {
+		return nil, "", err
 	}
-	stringDataMap := map[string]string{}
-	userPasswordSecret := secrets.NewSecret(namespacedName, secretName, stringDataMap, namer.LabelBuilder.Labels())
 
-	keyStorePassword := "password"
-	keyStorePath := "/etc/" + secretName + "-volume/broker.ks"
-	trustStorePassword := "password"
-	trustStorePath := "/etc/" + secretName + "-volume/client.ts"
-	if err := resources.Retrieve(secretNamespacedName, client, userPasswordSecret); err == nil {
-		if string(userPasswordSecret.Data["keyStorePassword"]) != "" {
-			keyStorePassword = string(userPasswordSecret.Data["keyStorePassword"])
-		}
-		if string(userPasswordSecret.Data["keyStorePath"]) != "" {
-			keyStorePath = string(userPasswordSecret.Data["keyStorePath"])
-		}
-		if string(userPasswordSecret.Data["trustStorePassword"]) != "" {
-			trustStorePassword = string(userPasswordSecret.Data["trustStorePassword"])
+	// for trust manager ca bundle.
+	// in user secret case it can have both keystore and truststore so a separate ca bundle is optional
+	var caSecret *corev1.Secret = nil
+	if caSecretName != nil {
+		trustSecretNamespacedName := types.NamespacedName{
+			Name:      *caSecretName,
+			Namespace: customResource.Namespace,
 		}
-		if string(userPasswordSecret.Data["trustStorePath"]) != "" {
-			trustStorePath = string(userPasswordSecret.Data["trustStorePath"])
+
+		caSecret = &corev1.Secret{}
+
+		if err := resources.Retrieve(trustSecretNamespacedName, client, caSecret); err != nil {
+			return nil, "", err
 		}
 	}
 
-	sslFlags = sslFlags + " " + "--ssl-key" + " " + keyStorePath
-	sslFlags = sslFlags + " " + "--ssl-key-password" + " " + keyStorePassword
-	sslFlags = sslFlags + " " + "--ssl-trust" + " " + trustStorePath
-	sslFlags = sslFlags + " " + "--ssl-trust-password" + " " + trustStorePassword
-	if customResource.Spec.Console.UseClientAuth {
-		sslFlags = sslFlags + " " + "--use-client-auth"
+	sslArgs, err := certutil.GetSslArgumentsFromSecret(sslSecret, keyStoreType, trustStoreType, caSecret, isConsole)
+	if err != nil {
+		return nil, "", err
 	}
 
-	return sslFlags
-}
-
-func generateAcceptorConnectorSSLArguments(customResource *brokerv1beta1.ActiveMQArtemis, namer common.Namers, client rtclient.Client, secretName string) string {
+	sslFlags := sslArgs.ToFlags()
 
-	sslArguments := "sslEnabled=true"
-	secretNamespacedName := types.NamespacedName{
-		Name:      secretName,
-		Namespace: customResource.Namespace,
-	}
-	namespacedName := types.NamespacedName{
-		Name:      customResource.Name,
-		Namespace: customResource.Namespace,
-	}
-	stringDataMap := map[string]string{}
-	userPasswordSecret := secrets.NewSecret(namespacedName, secretName, stringDataMap, namer.LabelBuilder.Labels())
+	if isConsole {
```

Comment:
I think it will make sense to take this logic out of this method, and not add an isConsole boolean argument. The caller can know about isConsole and deal with useClientAuth. That is not a common bit, I think from the renaming of the method that makes sense.

Reply:
+1